e2e/wdio/core/tests/table.e2e-spec.ts

```diff
@@ -0,0 +1,232 @@
+            saveElementScreenshot(tableCustomColumnsExample + table, 'table-custom-columns-example-' + getImageTagBrowserPlatform(),
+                tablePage.getScreenshotFolder());
+            expect(checkElementScreenshot(tableCustomColumnsExample + table, 'table-custom-columns-example-' + getImageTagBrowserPlatform(),
+                tablePage.getScreenshotFolder())).toBeLessThan(5, `element item state mismatch`);
+        });
+
+        it('should check search work correctly', () => {
+            scrollIntoView(tableCustomColumnsExample);
+            click(tableCustomColumnsExample + button);
+            setValue(dialogContent + inputGroup, 'date');
+            expect(getText(dialogValue)).toBe(dateTestText);
+        });
+    });
+
+    describe('Check Table with Checkboxes example', function() {
+
+        it('should check that checkbox work correctly', () => {
+            scrollIntoView(tableCheckboxesExample);
+            const checkboxLength = getElementArrayLength(tableCheckboxesExample + markAllCheckboxes);
+            for (let i = 0; i < checkboxLength; i++) {
+                scrollIntoView(tableCheckboxesExample + markAllCheckboxes, i);
+                click(tableCheckboxesExample + markAllCheckboxes, i);
+            }
+
+            const checkboxLength_2 = getElementArrayLength(tableCheckboxesExample + tableRow);
+            for (let i = 0; i < checkboxLength_2; i++) {
+                expect(getAttributeByName(tableCheckboxesExample + tableRow, 'aria-selected', i)).toBe('true');
+            }
+        });
+    });
+
+    describe('Check Table With Semantic Row Highlighting example', function() {
+
+        it('should check that checkbox work correctly', () => {
+            scrollIntoView(tableSemanticExample);
+            click(tableSemanticExample + markAllCheckboxes);
+            const checkboxLength = getElementArrayLength(tableSemanticExample + tableRow);
+            for (let i = 0; i < checkboxLength; i++) {
+                expect(getAttributeByName(tableSemanticExample + tableRow, 'aria-selected', i)).toBe('true');
+            }
+        });
+    });
+
+    describe('Check Pop In Mode on Responsive Table example', function() {
+
+        it('should check that table rows are clickable', () => {
+            scrollIntoView(tablePopinExample);
+            const tableRowLength = getElementArrayLength(tablePopinExample + clickableTableRow);
+            for (let i = 0; i < tableRowLength; i++) {
+                expect(isElementClickable(tablePopinExample + clickableTableRow, i)).toBe(true, `element with index ${i} not clickable`);
+            }
+        });
+
+        it('should check that checkbox work correctly', () => {
+            scrollIntoView(tablePopinExample);
+            click(tablePopinExample + markAllCheckboxes);
+            const checkboxLength = getElementArrayLength(tablePopinExample + tableRow);
+            for (let i = 20; i < checkboxLength; i++) {
```

Comment:
why 20? not obvious for debugging/reading. Maybe you can make a var with a name that will make it easy to understand?